### ISSUE TITLE
[question] efficiently simulating multiple SDE trajectories in d > 1

### ISSUE THREAD

**Issue by schwarzTass** [patrick-kidger/diffrax#531]
First of all many thanks for this great library. I've so far worked with the julia equivalent ([DifferentialEquations.jl](https://docs.sciml.ai/DiffEqDocs/stable/)) and find that this python library is closing a hole in the python ecosystem. Please forgive if I ask a diffrax-trivial question; I've tried my best with the given examples / github issue discussions.

I want to simulate a simple 2d SDE with additive noise, and I want to generate multiple paths. With my vmap attempt below, the paths are all the same; the BMs are not distinct, so my guess is that the way I create the BMs / seed them is not the proper way of doing it. Could you help me with fix this?

Second, for a 2d system of addtive noise, is using lx.DiagonalLinearOperator the best / most efficient way? Intuitively, I would have just returned a jnp.array of size 2, but that did not work.

```
import jax.random as jr
from diffrax import diffeqsolve, ControlTerm, Euler, MultiTerm, ODETerm, SaveAt, VirtualBrownianTree
from jax import numpy as jnp
import lineax as lx
import jax as jx
import time # for seeding

t0, t1 = 0, 1

# we define a simple 2d BM with drift
drift = lambda _, y, args: jnp.array([ -y[0], -y[1]])
diffusion = lambda _, y, args: lx.DiagonalLinearOperator(jnp.array([1,1])) # note that just returning jnp.array([1,1]) does not provide independent BMs

# the number of paths
n_paths = 10

# this is the function we will vmap
def solve(y0):
    # seed for this path
    seed = int(time.time_ns())
    print("seed:",seed)

    # first, create a BM for each path
    bm = VirtualBrownianTree(0, 1, tol=1e-3, shape=(2,), key=jr.PRNGKey(seed))

    # create the terms
    terms = MultiTerm(ODETerm(drift), ControlTerm(diffusion, bm))

    # solve the system
    return diffeqsolve(terms, Euler(), 0, 1, dt0=0.01, y0=y0, saveat=SaveAt(steps=True), max_steps=int(1e6))

# the initial condition
y0 = jnp.array([1.0, 2.0])
# we start always at this point
y0s = jnp.repeat(jnp.expand_dims(y0, axis=0), n_paths, axis=0)

# vmap the solve function
sol_multiple = jx.vmap(solve,in_axes=[0])
sols = sol_multiple(y0s)

print(sols.ys)
```

**Comment by johannahaffner** [comment]
Hi and welcome to the JAX side of things! Your issue is that you are vmapping over different values of 'y0', and JAX will define what operations to apply to each of them by looking at what operation to apply to one of them and then repeating that same operation on all. This means that it will only see one seed.

To fix this, define your seed outside of 'solve' and make it a vmapped argument, after splitting the key. 

Like so: 

```python 
def somefunction(value, key):
    ...

one_key = jr.key(one_seed)
many_keys = jr.split(key, n)

values = ...

results = jax.vmap(somefunction, in_axes=(0, 0))(values, keys)
```

This should result in different brownish paths.

**Comment by johannahaffner** [comment]
Hi and welcome to the JAX side of things! Your issue is that you are vmapping over different values of 'y0', and JAX will define what operations to apply to each of them by looking at what operation to apply to one of them and then repeating that same operation on all. This means that it will only see one seed.

To fix this, define your seed outside of 'solve' and make it a vmapped argument, after splitting the key. 

Like so: 

```python 
def somefunction(value, key):
    ...

one_key = jr.key(one_seed)
many_keys = jr.split(key, n)

values = ...

results = jax.vmap(somefunction, in_axes=(0, 0))(values, keys)
```

This should result in different brownish paths.

**Comment by johannahaffner** [comment]
Hi and welcome to the JAX side of things! Your issue is that you are vmapping over different values of 'y0', and JAX will define what operations to apply to each of them by looking at what operation to apply to one of them and then repeating that same operation on all. This means that it will only see one seed.

To fix this, define your seed outside of 'solve' and make it a vmapped argument, after splitting the key. 

Like so: 

```python 
def somefunction(value, key):
    ...

one_key = jr.key(one_seed)
many_keys = jr.split(key, n)

values = ...

results = jax.vmap(somefunction, in_axes=(0, 0))(values, keys)
```

This should result in different brownish paths.

**Comment by patrick-kidger** [comment]
For the additive noise question: in general 'additive noise' can refer to any kind of noise of the form `Σ(t) dW(t)`, where `W(t) ∈ R^n` and `Σ(t) ∈ R^nxm`.

But from your question it looks like you're specifically interested in the case that `Σ = [[1, 0], [0, 1]]` the constant 2x2 identity matrix?

If so then yup, you're doing looks correct! You could also use a `lineax.IdentityLinearOperator` if you wanted.

(Returning an array directly corresponds to defining `Σ`. We should maybe deprecate this, it's an easy way to make mistakes.)
